scripts/irods/test/test_iadmin.py

```diff
@@ -1823,6 +1823,41 @@ def test_iadmin_rmuser_removes_associated_acls__issue_7778(self):
             self.admin.run_icommand(["irmdir", test_collection_name])
             self.admin.run_icommand(["irm", "-f", test_dataobj_name])
 
+    def test_modzone_does_not_accept_invalid_zone_names__issue_7722(self):
+        # Show that "iadmin modzone name" no longer accepts invalid zone names.
+        invalid_zone_name = 'überZone'
+        ec, _, _ = self.admin.assert_icommand(
+            ['iadmin', 'modzone', self.admin.zone_name, 'name', invalid_zone_name],
+            'STDERR',
+            ['-130000 SYS_INVALID_INPUT_PARAM'],
+            input='yes')
+        self.assertNotEqual(ec, 0)
```

Comment:
Do we expect the same ec every time from the `iadmin ...` command? If we do, we should use `desired_rc` in the `assert_icommand(...)` function.

This may catch changes that may break expectations (e.g. scripting).

Reply:
The actual error code returned to the OS by the icommands aren't important. Unless otherwise stated, their only purpose is to indicate 0 (success) or non-zero (failure).